typescript/sdk/src/providers/transactions/submitter/ethersV5/ImpersonatedAccountTxSubmitter.ts

```diff
@@ -0,0 +1,66 @@
+import { ContractReceipt } from 'ethers';
+import { Logger } from 'pino';
+
+import { rootLogger } from '@hyperlane-xyz/utils';
+import { Address } from '@hyperlane-xyz/utils';
+
+import { ChainName } from '../../../../types.js';
+import { impersonateAccount } from '../../../../utils/fork.js';
+import { MultiProvider } from '../../../MultiProvider.js';
+import {
+  EthersV5Transaction,
+  EthersV5TransactionReceipt,
+  ProviderType,
+} from '../../../ProviderType.js';
+import { TxSubmitterInterface } from '../TxSubmitter.js';
+import { TxSubmitterType } from '../TxSubmitterTypes.js';
+
+interface ImpersonatedAccountTxSubmitterProps {
+  address: Address;
+}
+
+export class ImpersonatedAccountTxSubmitter
+  implements
+    TxSubmitterInterface<EthersV5Transaction, EthersV5TransactionReceipt>
+{
+  public readonly txSubmitterType: TxSubmitterType =
+    TxSubmitterType.IMPERSONATED_ACCOUNT;
+
+  protected readonly logger: Logger = rootLogger.child({
+    module: 'impersonated-account-submitter',
+  });
+
+  constructor(
+    public readonly multiProvider: MultiProvider,
+    public readonly chain: ChainName,
+    public readonly props: ImpersonatedAccountTxSubmitterProps,
+  ) {}
+
+  public async submitTxs(
+    txs: EthersV5Transaction[],
+  ): Promise<EthersV5TransactionReceipt[]> {
+    const receipts: EthersV5TransactionReceipt[] = [];
+    for (const tx of txs) {
+      const receipt = await this.submitTx(tx);
+      receipts.push(receipt);
+    }
+    return receipts;
+  }
+
+  public async submitTx(
+    tx: EthersV5Transaction,
+  ): Promise<EthersV5TransactionReceipt> {
+    const signer = await impersonateAccount(this.props.address);
+    this.multiProvider.setSigner(this.chain, signer);
+    const receipt: ContractReceipt = await this.multiProvider.sendTransaction(
+      this.chain,
+      tx.transaction,
+    );
```

Comment:
should we inherit this behavior from JsonRPC submitter?

Reply:
Thought through this a bit more– and I have decided to extend this by `EV5JsonRpcTxSubmitter`. Now, `EV5ImpersonatedAccountTxSubmitter`'s submit() calls to its parent's function after impersonation